### PR TITLE
remove unreachable code and if-clauses which evaluate always to true/false

### PR DESCRIPTION
**Pull request by rabauke**
I applied some static code analysis tools to the darktable sourcs and found, among others, unreachable code and if-clauses which evaluate always to true/false.

### CODE REVIEW

File: src/common/imageio_rawspeed.cc [hunk 1]
```diff
@@ -399,7 +398,7 @@ dt_imageio_retval_t dt_imageio_open_rawspeed_sraw(dt_image_t *img, RawImage r, d
       }
     }
   }
-  else if(cpp == 3 || cpp == 4)
+  else
```
Comment (TurboGit) [inline src/common/imageio_rawspeed.cc]: ok, but I would keep the comment for the readers:

   else // case cpp == 3 or 4

File: src/libs/live_view.c [hunk 2]
```diff
@@ -524,13 +524,7 @@ void gui_post_expose(dt_lib_module_t *self, cairo_t *cr, int32_t width, int32_t
...
     if(!img) img = dt_image_cache_get(darktable.image_cache, imgid, 'r');
 
-    int zoom = 1;
-    float imgwd = 0.90f;
-    if(zoom == 1)
-    {
-      imgwd = .97f;
-    }
-
+    float imgwd = 0.97f;
```
Comment (TurboGit) [inline src/libs/live_view.c]: Indeed, looks like a code after refactoring :)

Comment (TurboGit) [reply src/libs/live_view.c]: And as we are at it, make imgwd a const to make it clear that it does not change.

**Comment by rabauke** [comment]
Requested changes applied, plus some further minor fixes.

File: src/iop/clipping.c [hunk 3]
```diff
@@ -1187,12 +1187,6 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
     d->all_off = 1;
     d->k_apply = 0;
   }
-  else
```
Comment (TurboGit) [inline src/iop/clipping.c]: Are you sure about this one?

We have:

```
if(p->k_type == 4)
else if(p->k_type >= 0 && p->k_apply == 1)
else if(p->k_type == 0)
else
```

So the else part can be taken if p->k_type > 0 and p->k_apply == 0.

Comment (upegelow) [reply src/iop/clipping.c]: Out of own experience I'd be very cautious with any changes to that module shortly before a release. IMHO it's better to allow for intensive testing in the development branch.

**Comment by TurboGit** [comment]
Merged manually. Better sooner than later, but this new version seems correct to me. Thanks.

**Pull request closed without merging**